### ISSUE TITLE
Improve UX of nomad deployment API and CLI

### ISSUE THREAD

**Issue by wuub** [hashicorp/nomad#3157]
### Nomad version
0.6.2

### Operating system and Environment details
n/a

### Issue

Nomad deployment API & cli lacks some critical data and functions from the usability/UX standpoint.

1. `nomad status` shows job's submit date, `nomad deployment list` does not. Internally it doesn't even seem to track deployment creation time, just raft index of create/modify. 
2. There is no easy way to filter or page deployment objects. Filtering by at least status or creation date (see 1.) would be interesting.
3. Number of deployments seems to grow, after just a few days with 0.6.2 we're at 400+ lines output of `nomad deployment list`. Are deployment objects garbage collected? When?  
4. Jobs without `update {}` block do not seem to create a deployment object. Why is that? It's a bit confusing. 
5. Being able to use `nomad deployment status [job-name]` instead of deployment `uuid` to see the last one would be great improvement. 

EDIT: 

6. We often find ourselves looking at only the last deployment for a job. Unless I'm doing a deep investigation on job's history, I only care about the status of the last one. Ability to easily inspect the status of last deployment of ~specific~ (EDIT: ok, `nomad job deployments -latest` does what I need)  /all ( for consistency's sake `nomad deployment list -latest` would probably be ok)
jobs would be nice.
7. Regarding the above. If nomad starts doing GC of deployment objects, I think the GC should not be strictly time/index based, but instead do a Kafka'like log compaction, always leaving at least one deployment object for a job.

**Comment by dansteen** [comment]
Just to throw my two cents in,  I'd love to see a bit more consistency around some of the CLI options in general.   As an example we have the following:

```
job status 
job list
deployment status 
deployment list
```

But we don't have:

``
alloc status
alloc list
eval status
eval list
```
And while we do have `alloc-status` and eval-status` those are hyphenated rather than subcommands.  Even if it doesn't make sense to provide `eval list` and `alloc list` (and I'd actually like to see those), it would be nice if the interface was consistent and had `eval status` as a subcommand.

Even among the existing subcommands there are inconsistencies that throw you.  As examples
1.  `job status` will list jobs, but `deployment status` will throw an error.   
2. `job status <name>` will give the status of the job named <name>, but `deployment status <name>` will throw an error because it expects a UUID (since deployments are tied to jobs why not just list the deployments for that job?)

In 0.6.2 an attempt was made to fix some of this by providing the `status` command,  but I think this really just made things worse:
1. the `status` command does not accept all of the options of each of the dedicated `<item> status` (or `<item>-status`) commands.
2. Prior to 0.6.2 running `status <job name>` would give you the status of <job name> but now it throws an error since it's expecting a UUID

All this adds a lot of cognitive load to using the CLI, and makes it really hard for new people to pick it.  I love CLI tools, and almost always prefer them over GUIs, but I find myself using the Web-UI to nomad when I really need to debug something since it provide the information in a more consumable and consistent form.  :-(

**Comment by dadgar** [comment]
@dansteen We are moving exactly in the direction you stated. It just takes time 👍 The status command returning an error based on uuid is a bug that will be fixed in 0.6.3 which is out in RC right now so play with it 😄

**Comment by dansteen** [comment]
@dadgar no problem!   I can definitely be patient for this.  Just figured I'd through in some thoughts
